CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -30,6 +31,8 @@ internal static class HtmlFormatterSlim
             true,  true,  true,  true,  true,  true,  true,  true,  true,  true,  true,  false, false, false, false, false
         };
 
+        private static readonly Dictionary<string, uint> GitHubHeaderIdCounts = new Dictionary<string, uint>();
```

Comment:
you can make it thread safe by not creating a shared static field that uses non thread safe dictionary. this simply will not produce correct results.